### ISSUE TITLE
TDD: Greedy action mode

### ISSUE THREAD

**Issue by LSailer** [LSailer/Master-Thesis-3D-VLA#23]
## TDD Cycle 2: Greedy action mode

**Feature:** `agent.act(obs, key, training=False)` uses `jnp.argmax` instead of sampling

**Spec ref:** `docs/specs/dreamerv3-objectnav-baseline.md` §7, Cycle 2

### Test (`tests/test_dreamerv3_shapes.py`)

```python
class TestGreedyAction:
    def test_greedy_is_deterministic(self, cfg, rng):
        """training=False → argmax → same action for same state."""
        from src.dreamerv3.agent import DreamerAgent
        agent = DreamerAgent(cfg, rng)
        obs = {"image": jnp.zeros(cfg.obs_shape, dtype=jnp.uint8), "is_first": True}
        k1, k2 = jax.random.split(rng)
        a1 = agent.act(obs, k1, training=False)
        a2 = agent.act(obs, k2, training=False)
        assert a1 == a2, "greedy mode must be deterministic regardless of rng key"

    def test_stochastic_uses_sampling(self, cfg, rng):
        """training=True → categorical sampling (default behavior)."""
        import inspect
        from src.dreamerv3.agent import DreamerAgent
        source = inspect.getsource(DreamerAgent._act_forward)
        assert "training" in inspect.signature(DreamerAgent._act_forward).parameters or \
               "argmax" in source or "categorical" in source
```

### Implementation (`src/dreamerv3/agent.py`)

- Pass `training` through `act()` → `_act_jit` → `_act_forward`
- In `_act_forward`: `action = jnp.where(training, jax.random.categorical(rng_key, logits), jnp.argmax(logits, axis=-1))`
- Simplest: use `functools.partial` with `static_argnums` for `training` flag

### Pass command
```bash
uv run pytest tests/test_dreamerv3_shapes.py::TestGreedyAction -v
```

### Commit message
`add greedy action mode (training=False → argmax)`

### Files
- `src/dreamerv3/agent.py` (modify)
